lib/searchkick/query.rb

```diff
@@ -490,10 +492,26 @@ def set_fields
           []
         elsif default_match == :exact
           raise ArgumentError, "Must specify fields to search"
+        elsif has_field_misspellings(options[:misspellings])
+          raise ArgumentError, "If you provide per-field misspelling rules you must also provide the fields to search."
         else
           [default_match == :word ? "*.analyzed" : "*.#{default_match}"]
         end
-      [boost_fields, fields]
+      [boost_fields, fields, option_fields_mapping]
+    end
+
+    def has_field_misspellings(misspellings)
+      misspellings.is_a?(Hash) &&
+        misspellings[:fields].is_a?(Hash) &&
+        misspellings[:fields].keys.any?
+    end
+
+    def misspellings_for_field(misspellings, field)
+      if misspellings.is_a?(Hash) && misspellings[:fields] && misspellings[:fields][field] != nil
+        misspellings[:fields][field]
```

Comment:
Think we should merge this with `misspellings` if it's a hash, so `misspellings: {edit_distance: 2, fields: {color: {transpositions: false}}` will keep the edit distance for color.

Reply:
:+1: updated in 15cc52b58fe0c7127856381d712e78d8c08befb0